### PR TITLE
fix(race-issue): docker race issue

### PR DESCRIPTION
**Pull request by bishnubista**
Wait for one app to start before returning

## **Description**
Instead of spawning the one app process, then exiting, wait for it to definitely start.

## **Motivation** 
Since the current code does not explicitly wait for one app to start, processes that rely on this function sometimes start trying to contact one app before it is ready.

## **Test** **Conditions**
Unit tests

#### Check boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

#### Check boxes that apply:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: packages/one-app-runner/bin/one-app-runner-test.js [hunk 1]
```diff
@@ -16,6 +16,23 @@
...
-spawn(command, { shell: true, stdio: 'ignore', detached: true }).unref();
+
+const run = async () => {
+  spawn(command, { shell: true, stdio: 'ignore', detached: true }).unref();
+  const port = process.env.HTTP_PORT;
+  const timeout = 200000;
+
+  console.log(`Waiting for one app to start on port ${port}`);
+
+  const val = await waitForOK({
```
Comment (giulianok) [inline packages/one-app-runner/bin/one-app-runner-test.js:30]: not a blocker but it would be nice to have a better name for `val`, doesn't tell me much

**Review by giulianok — APPROVED** [review]

**Review by dogpatch626 — APPROVED** [review]

File: packages/one-app-runner/bin/one-app-runner-test.js (hunk 1 above)
Comment (bishnubista) [reply packages/one-app-runner/bin/one-app-runner-test.js]: Agreed. I have changed to `resolvedStatus`.

**Review by dogpatch626 — APPROVED** [review]

File: packages/one-app-runner/bin/one-app-runner-test.js [hunk 2]
```diff
@@ -16,6 +16,23 @@
...
+  const timeout = 200000;
+
+  console.log(`Waiting for one app to start on port ${port}`);
+
+  const resolvedStatus = await waitForOK({
+    url: `http://localhost:${port}/_/status`,
+    timeout,
+  });
+
+  console.log(`${resolvedStatus ? 'One app server started successfully' : `One app crashed in ${timeout}ms`}`);
```
Comment (10xLaCroixDrinker) [inline packages/one-app-runner/bin/one-app-runner-test.js:35]: Please exit if it crashes

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]

File: packages/one-app-runner/bin/one-app-runner-test.js (hunk 2 above)
Comment (bishnubista) [reply packages/one-app-runner/bin/one-app-runner-test.js]: Done.

**Review by giulianok — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**